server/src/main/java/org/elasticsearch/index/query/MultiMatchQueryBuilder.java

```diff
@@ -534,10 +539,20 @@ public MultiMatchQueryBuilder zeroTermsQuery(MatchQuery.ZeroTermsQuery zeroTerms
         return this;
     }
 
+    public MultiMatchQueryBuilder synonymQueryStyle(MatchQuery.SynonymQueryStyle synQueryStyle) {
+        if (synonymQueryStyle == null) {
+            throw new IllegalArgumentException("[" + NAME + "] requires synonym query style to be non-null");
```

Comment:
Here it is more clean, but again I think using `synonym_query_style` would be better